### PR TITLE
Sending emails is done through vs-api

### PR DESCRIPTION
**Pull request by akbarik**
### Related issues

https://github.com/DivanteLtd/vue-storefront/issues/1812

### Short description and why it's useful

There's a new endpoint in vs-api that send emails using nodemailer library. Now frontend doesn't rely on customer's mail application, it just sends data to vs-api which handles email sending. New module called "mailer" has been created for this. In config file one can define the email address and need for confirmation.

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

### Contribution and curently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and [refactoring plan for them](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/refactoring-to-modules.md)

### CODE REVIEW

**Comment by pkarw** [comment]
Thanks @akbarik! We just need to make it more secure - please check the comment in the API PR

**Comment by filrak** [comment]
Hey, @akbarik can you adjust it to new structure of modules? https://github.com/DivanteLtd/vue-storefront/pull/1798 ? sorry if it wasn't clear but I'm currently refactoring all the previous ones. From what i see it should be straightforward

**Comment by filrak** [comment]
@akbarik I updated docs about modules with some useful info (rules and best practices) https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md -  may be useful

**Comment by akbarik** [comment]
Thanks, @filrak! I'll refactor the code accordingly then.

File: core/modules/mailer/features/sendEmail.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import config from 'config'
```
[the remark this answers is not in the document]
Comment (filrak) [reply core/modules/mailer/features/sendEmail.ts]: also would be great if you'll provide some validation

**Comment by filrak** [comment]
Also mwe should have option to enter email we should respond to ;)

**Comment by akbarik** [comment]
> Also we should have option to enter email we should respond to ;)

The Mailer module has a sourceAddress param where you can pass whatever email you would like to respond to. But in this specific use case, when we are sending a feedback after checkout, I think it's not needed and, in my opinion, will not be very UX-friendly to ask user to fill in email address, since we already asked him to at the checkout. I'm not sure though, what's the probability of user would want to receive a confirmation on a different email address..

**Comment by pkarw** [comment]
is it ready to review and merge @akbarik ?

**Comment by akbarik** [comment]
Yes, @pkarw, it's ready. Please let me know if I still missed something.

File: core/modules/mailer/features/sendEmail.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (akbarik) [reply core/modules/mailer/features/sendEmail.ts]: @filrak, can you clarify what kind of validation? Like checking the email format and emptiness of string values?

**Pull request merged**